### PR TITLE
 macvlan, bridge, ptp: send gratuitous arp, don't change HWAddr

### PR DESCRIPTION
**Pull request by squeed**
We used to set the HWAddr based on the assigned IP in case we re-use IPs in a short timescale, so we don't have to wait for the arp table to time out. Since this makes ipv6 harder, don't do it. Send a gratuitous ARP instead.

Fixes: #438

### CODE REVIEW

**Comment by dcbw** [comment]
LGTM, though major bonus points if you use gopacket to detect and verify that the ARP gets sent :)

http://www.devdungeon.com/content/packet-capture-injection-and-analysis-gopacket

**Comment by squeed** [comment]
Also for the record: we added round-robin to the `host-local` ipam plugin, so the chance of quickly re-using an IP is a lot lower than before.

**Comment by squeed** [comment]
Ask and ye shall receive, @dcbw  :-)

**Comment by dcbw** [comment]
Neato, bonus approved and in the mail.

**Comment by bboreham** [comment]
Wondering whether "github.com/j-keck/arping" would be a better dependency; it seems to have a smaller footprint.

**Comment by squeed** [comment]
@bboreham I think you're right - I'll swap it out. Shouldn't change much.

**Comment by squeed** [comment]
@bboreham Swapped libraries and updated. Ahh, much better.

**Comment by dcbw** [comment]
LGTM

**Comment by squeed** [comment]
This probably shouldn't be merged - Once it gets another LGTM, I'll move it to the plugins repository.

**Comment by squeed** [comment]
Nevermind, we should merge this and then snarf that commit in to plugins.

We should also delete the plugins here soon :-)

**Comment by bboreham** [comment]
> Also, added some test deps to godeps.json, but didn't change any actual vendor files

Can you explain this a bit more?  Why do we want to list things in godeps.json that we don't actually vendor in?

File: pkg/ip/arp.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"net"
+
+	"github.com/j-keck/arping"
+)
+
+// GratuitousArp sends an arp for the supplied IP address on the interface ifName
+func GratuitousArp(iface *net.Interface, addr net.IP) error {
+	return arping.GratuitousArpOverIface(addr, *iface)
```
Comment (bboreham) [inline pkg/ip/arp.go]: I question the value of a 26-line file and 99-line test to swap two arguments round

**Comment by bboreham** [comment]
Aside from complaining about the function that swaps two arguments round, the basic functionality lgtm.

**Comment by squeed** [comment]
Closing in favor of containernetworking/plugins#16

**Pull request closed without merging**